From 34a770dea89a79b3548d1ebaf1591ab76aaf70dd Mon Sep 17 00:00:00 2001
From: Edwin Peer <edwin.peer@netronome.com>
Date: Fri, 21 Sep 2018 17:05:37 -0700
Subject: [PATCH 3/9] [PCI.OUT] optimize ticket release in PD

The zero result from ticket release can be safely added with no functional
effect. This cycle improvement comes at the cost of additional load on the
MU atomic engine. Empirical testing demonstrates that the PD ME is cycle
bound and better overall performance is achieved by making this tradeoff.

Signed-off-by: Edwin Peer <edwin.peer@netronome.com>
---
 me/blocks/vnic/pci_out_pd.uc | 1 -
 1 file changed, 1 deletion(-)

diff --git a/me/blocks/vnic/pci_out_pd.uc b/me/blocks/vnic/pci_out_pd.uc
index b96ea22b..828283a0 100644
--- a/me/blocks/vnic/pci_out_pd.uc
+++ b/me/blocks/vnic/pci_out_pd.uc
@@ -812,7 +812,6 @@ no_ctm_buffer#:
     mem[fast_journal, --, g_blm_addr_hi, <<8, addr_lo], indirect_ref
 
 ticket_ready#:
-    br=byte[$ticket, 0, 0, complete_done#]
     br=byte[$ticket, 0, TICKET_ERROR, ticket_error#]
 
     alu[--, g_add_imm_iref, OR, $ticket, <<16]
-- 
2.19.0

